eagle-core/eagle-app/eagle-app-base/src/main/java/org/apache/eagle/app/service/impl/ApplicationManagementServiceImpl.java

```diff
@@ -63,16 +71,40 @@ public ApplicationEntity install(ApplicationOperations.InstallOperation operatio
         ApplicationEntity applicationEntity = new ApplicationEntity();
         applicationEntity.setDescriptor(appDesc);
         applicationEntity.setSite(siteEntity);
-        applicationEntity.setConfiguration(operation.getConfiguration());
+
+        /**
+         *  calculate application config based on
+         *   1) default values in metadata.xml
+         *   2) user's config value
+         *   3) some metadata, for example siteId, mode, appId
+         */
+        Map<String, Object> appConfig = new HashMap<>();
+        ApplicationProvider provider = applicationProviderService.getApplicationProviderByType(operation.getAppType());
+        List<Property> propertyList = provider.getApplicationDesc().getConfiguration().getProperties();
+        for(Property p : propertyList){
+            appConfig.put(p.getName(), p.getValue());
+        }
+        if(operation.getConfiguration() != null) {
+            appConfig.putAll(operation.getConfiguration());
+        }
+        appConfig.put("siteId", operation.getSiteId());
+        appConfig.put("mode", operation.getMode().name());
+        appConfig.put("appId", operation.getAppType());
+
+        applicationEntity.setConfiguration(appConfig);
         applicationEntity.setMode(operation.getMode());
         ApplicationContext applicationContext = new ApplicationContext(
                 applicationProviderService.getApplicationProviderByType(applicationEntity.getDescriptor().getType()).getApplication(),
                 applicationEntity,config);
         applicationContext.onInstall();
         applicationEntityService.create(applicationEntity);
+
+        // initialize web config
+        provider.installWebConfig(ConfigFactory.parseMap(appConfig));
```

Comment:
Not sure the difference between `installWebConfig ` with `applicationEntity.setConfiguration(appConfig);` ?